main_test.go

```diff
@@ -32,3 +33,29 @@ func TestListen(t *testing.T) {
 	_, err = os.Create(endpoint)
 	require.NoError(t, err)
 }
+
+func TestSetupLogger(t *testing.T) {
+	tests := []struct {
+		flags    config.FlagsConfig
+		expected hclog.Level
+	}{
+		{config.FlagsConfig{Debug: true}, hclog.Debug}, // deprecated flag test
```

Comment:
Might be good to test the case where both `Debug` and `LogLevel` are set.

Reply:
Good idea. And this brings up a good point too - Do we want the `LogLevel` to take precedence if they are both set?
Currently, if the `Debug` flag is set to true and `LogLevel` is set to `warn` for example, the `Debug` flag will take precedence. This is how the `setupLogger` func is right now. Do we want to keep it like this, or change our `else` block. Do we have a preference?

Reply:
I'd probably expect `LogLevel` to take precedence now that you mention it, but I don't have a strong preference.

Reply:
I agree, I think that LogLevel should take precedence. Updated in https://github.com/hashicorp/vault-csi-provider/pull/295/commits/eb12344a3d36e8d6b16efb39990f5d896157c7cf